web/views.py

```
@@ -6961,6 +6979,451 @@ def delete_post(request, post_id):
+    """Cancel the user's subscription at the end of the billing period."""
+    if not hasattr(user, "membership") or not user.membership.stripe_subscription_id:
+        return {"success": False, "error": "No active subscription found"}
+
+    try:
+        setup_stripe()
+        subscription = stripe.Subscription.modify(user.membership.stripe_subscription_id, cancel_at_period_end=True)
+
+        # Update local membership data
+        update_membership_from_subscription(user, subscription)
+
+        return {"success": True}
```

Ruff (0.8.2)

Consider moving this statement to an `else` block

(TRY300)